### PR TITLE
android/ui: support receiving Taildrop files directly to Downloads fo…

### PR DESCRIPTION
**Pull request by oxtoacart**
…lder

Closes tailscale/tailscale#11705

### CODE REVIEW

File: android/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
     <uses-permission
         android:name="android.permission.WRITE_EXTERNAL_STORAGE"
-        android:maxSdkVersion="28" />
+        android:maxSdkVersion="29" />
```
Comment (oxtoacart) [inline android/src/main/AndroidManifest.xml:10]: This makes sure we also request this permission on Android 10

File: android/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -28,6 +28,7 @@
         android:name=".App"
         android:allowBackup="false"
         android:banner="@drawable/tv_banner"
+        android:requestLegacyExternalStorage="true"
```
Comment (oxtoacart) [inline android/src/main/AndroidManifest.xml:31]: We need this in order to be able to use `Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS)` on Android 10.

File: android/src/main/java/com/tailscale/ipn/App.kt [hunk 3]
```diff
@@ -317,25 +315,6 @@ class App : Application(), libtailscale.AppContext {
       return null
     }
 
-  @Throws(IOException::class)
```
Comment (oxtoacart) [inline android/src/main/java/com/tailscale/ipn/App.kt:320]: Unused vestigial code related to Taildrop.

**Comment by barnstar** [comment]
The issue ID in the commit appears to be incorrect...

**Comment by barnstar** [comment]
> The issue ID in the commit appears to be incorrect...

tailscale/tailscale#11705

**Comment by oxtoacart** [comment]
I tested this on an Android 10 emulator and verified the following:

1. Without this fix, we do in fact encounter the OG [issue](https://github.com/tailscale/tailscale/issues/11705)
2. With this fix, the permission prompt works as expected
3. With this fix, files that are Taildropped to the Android device successfully make it into the `Downloads` folder

I had originally hoped to be able to just not request this permission on Android 10, but I tried that and Taildrop was unable to save incoming files to `Downloads`.

**Review by barnstar — APPROVED** [review]

**Review by barnstar — APPROVED** [review]
🚀

**Pull request merged**